dev-tools/omdb/src/bin/omdb/db.rs

```diff
@@ -1352,6 +1395,288 @@ async fn cmd_db_eips(
+        let client = CrucibleAgentClient::new(&url);
+
+        let actual_region_snapshots = client
+            .region_get_snapshots(&RegionId(
+                region_snapshot.region_id.to_string(),
+            ))
+            .await?;
+
+        // Print a message if the Agent says that the snapshot was deleted
+
+        let snapshot_id = region_snapshot.snapshot_id.to_string();
+
+        if actual_region_snapshots
+            .snapshots
+            .iter()
+            .any(|x| x.name == snapshot_id)
+        {
+            // A snapshot currently exists, matching the database entry
+        } else {
+            // In this branch, there's a database entry for a snapshot that was
+            // deleted. Due to how the snapshot create saga is currently
+            // written, a database entry would not have been created unless a
+            // snapshot was successfully made: unless that saga changes, we can
+            // be reasonably sure that this snapshot existed at some point.
+
+            match actual_region_snapshots.running_snapshots.get(&snapshot_id) {
+                Some(running_snapshot) => {
+                    match running_snapshot.state {
+                        State::Destroyed | State::Failed => {
+                            // In this branch, we can be sure a snapshot previously
+                            // existed and was deleted: a running snapshot was made
+                            // from it, then deleted, and the snapshot does not
+                            // currently exist in the list of snapshots for this
+                            // region. This record should be deleted.
+
+                            // Before recommending anything, validate the higher
+                            // level Snapshot object too: it should have been
+                            // destroyed.
+
+                            let snapshot: Snapshot = {
+                                use db::schema::snapshot::dsl;
+
+                                dsl::snapshot
+                                    .filter(
+                                        dsl::id.eq(region_snapshot.snapshot_id),
+                                    )
+                                    .select(Snapshot::as_select())
+                                    .first_async(
+                                        &*datastore
+                                            .pool_connection_for_tests()
+                                            .await?,
+                                    )
+                                    .await?
+                            };
+
+                            if snapshot.time_deleted().is_some() {
+                                // This is ok - Nexus currently soft-deletes its
+                                // resource records.
+
+                                eprintln!(
+                                    "region snapshot {} {} {} at {} was deleted, please remove its record",
```

Comment:
This one could use a header for DATASET REGION SNAPSHOT or some such to describe these three.

Seeing as the pattern in repeated below, but just the error changes, you could probably always print the triplet, then just customize the message depending on what the error was.

Reply:
agreed, changed in 8b1200535